cypress/integration/core_opensearch_dashboards/opensearch_dashboards/apps/explore/03/shared_links.spec.js

```diff
@@ -187,18 +179,18 @@ export const runSharedLinksTests = () => {
           openShareMenuWithRetry();
           cy.getElementByTestId('exportAsSavedObject').find('input').should('not.be.disabled');
```

Comment:
is the element actually called export as saved object? is it the same as discover? if so its confusing because we have the export saved object feature where as this i think it creates a saved object . or perhaps this could have a better test name describing what it does?

because shared link as snapshot could technically still have the saved object id if we created a shared link from a saved explore. but i think the difference between the two would be it will just load the saved object whereas a snapshot could be the saved object + current state. does this test end up passing even if state is included? same for snapshot test

Reply:
yea this test only tests scenarios:
1. snapshot link captures current state
2. saved object link shared the saved explore 

it doesn't test: 
after we save as saved explore and we make changes and we share:
1. snapshot link should include the changes
2. saved object link should not include the unsaved changes

I could add a todo to add tests on the latter two @kavilla